### DISCUSSION TITLE
Exploring and Visualizing Culture Homework

### DISCUSSION THREAD

**Discussion by ZoeLeBlanc** [CultureAsData-UIUC/is310-fall-2024#11] [Assignments]
Please post a link to your folder or notebook for the Exploring and Visualizing Culture Homework, details available here [https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/interpreting-communicating-humanities-data/02-eda-data-viz#exploring-and-visualizing-culture-homework](https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/interpreting-communicating-humanities-data/02-eda-data-viz#exploring-and-visualizing-culture-homework).

**Comment by haydnelm** [comment]
https://github.com/haydnelm/is310-coding-assignments/tree/main/pandas-eda

**Comment by GloriaLiu930** [comment]
https://github.com/GloriaLiu930/is310-coding-assignments/tree/main/pandas-eda

**Comment by ryanpearlman1** [comment]
https://github.com/ryanpearlman1/is310-coding-assignments/tree/main/pandas-eda

**Comment by divyaj5** [comment]
https://github.com/divyaj5/is310-coding-assignments/tree/master/pandas-eda

**Comment by tinayang922** [comment]
https://github.com/tinayang922/is310-coding-assignments/tree/main/pandas-eda

**Comment by averyWebster** [comment]
https://github.com/averyWebster/is310-coding-assignments/tree/main/pandas-eda

**Comment by frangara** [comment]
https://github.com/frangara/is310-coding-assignments/tree/main/pandas-eda

**Reply by jrfrye2** [reply]
Hi @frangara 

There are no problems with your code, but you did not include an analysis of the data and your findings. So that others can better understand your argument, you should provide an explanation of your goals and an interpretation of the visualizations. 

It was also unclear if Top500NovelsEDA was a part of your assignment or if I should only be looking at NYTBESTsellersRankandTimeEDA, so clearer documentation is needed but otherwise no issues.

**Comment by shoirathi** [comment]
https://github.com/shoirathi/is310-coding-assignments/tree/main/pandas-eda

**Reply by jrfrye2** [reply]
Hi @shoirathi 

Your code runs fine, but you did not include a README.

**Comment by henRotunno** [comment]
https://github.com/henRotunno/is310-coding-assignments/tree/main/pandas-eda

**Comment by LucasWashor** [comment]
https://github.com/LucasWashor/is310-coding-assignments/tree/main/pandas-eda

**Reply by jrfrye2** [reply]
Hi @LucasWashor 

Your third visualization, "Popularity Over Time" returns this error: 

![Screenshot 2024-12-06 190608](https://github.com/user-attachments/assets/8a0d89b1-0ef9-4759-81dd-dec41cb070d4)


After fixing that, all the other code looks good!

**Comment by OctaneCodes** [comment]
https://github.com/OctaneCodes/is310-coding-assignments/tree/main/pandas-eda

**Reply by jrfrye2** [reply]
Hi @OctaneCodes 

All your code works fine, but you did not include clear labels for each section or interpretation and discussion of your visualizations.

**Comment by SamuelSokolovsky** [comment]
https://github.com/SamuelSokolovsky/is310-coding-assignments/tree/main/pandas-eda/first_eda

**Reply by jrfrye2** [reply]
Hi @SamuelSokolovsky 

Your genre counts visualization did not show up for me: 
![Screenshot 2024-12-06 203554](https://github.com/user-attachments/assets/2a5aad19-d695-48e0-a943-5124312b18fb)

You could have also made your section headings more clear, but other than that it looks good!

**Comment by JBrown-0** [comment]
https://github.com/JBrown-0/is310-coding-assignments/tree/main/pandas-eda

**Reply by jrfrye2** [reply]
Hi @JBrown-0 

Your code works, but you did not do any visualizations with Altair. There are also no labels, and you did not do any analysis or discussion of the data. It would be especially helpful to explain what the visualizations are trying to tell us. That would not be immediately clear to someone unfamiliar with the dataset. 

Your theme was genre and publication trends, but the visualizations you provided do not illustrate the relationship between these two things.

**Comment by lhayedimarucut** [comment]
https://github.com/lhayedimarucut/is310-coding-assignments/tree/main/pandas-eda

**Comment by Cindy1226** [comment]
https://github.com/Cindy1226/is310-coding-assignments/tree/main/pandas-eda

**Reply by jrfrye2** [reply]
Hi @Cindy1226 

Your scatterplot gave me this output with most of the data missing. 

![Screenshot 2024-12-07 163635](https://github.com/user-attachments/assets/886edbcd-d417-47a2-854c-cbfd1b05ec08)

You also did not use Altair for any of your visualizations. 
Other than that, everything works and your organization and discussion look good!

**Reply by Cindy1226** [reply]
Thank you for the feedback! I replaced the previous scatterplot with a new visualization using Altair, as the original scatterplot wasn't working properly even after I attempted to fix it.  https://github.com/Cindy1226/is310-coding-assignments/tree/main/pandas-eda

**Comment by Rebe-Rod** [comment]
https://github.com/Rebe-Rod/is310-coding-assingments/blob/main/pandas-eda/GenresAndRanks.ipynb

**Reply by Rebe-Rod** [reply]
Sorry that's the wrong link. Here is the correct link! 
https://github.com/Rebe-Rod/is310-coding-assingments/blob/main/pandas-eda/GenreAndAvgRanks.ipynb

**Reply by jrfrye2** [reply]
Hi @Rebe-Rod 

Your pie chart gave me this error: 
![Screenshot 2024-12-06 212057](https://github.com/user-attachments/assets/52d87428-bd88-4f7f-af31-d43a1c8d9d3a)

Aside from that one error, all the code, formatting, and analysis looks good!

**Comment by yosefstifanos** [comment]
https://github.com/yosefstifanos/is310-coding-assignments/tree/main/pandas-eda

**Reply by jrfrye2** [reply]
Hi @yosefstifanos 

Your code works, but you did not include any discussion or analysis of the data. Your file name does not describe the project, and you only included one visualization. 

Your README says that you planned to analyze soccer-themed literature, but this theme is not explored in your code.

**Reply by yosefstifanos** [reply]
@jrfrye2 Sorry about the incorrect title. Ideally I wanted to do soccer, but then I changed my mind midway and forgot to change the README. I will redo this and then ping you for a regrade.

**Reply by yosefstifanos** [reply]
@jrfrye2 just refinished it again. lmk if I need to do anything else.

**Comment by theajk2** [comment]
htpps://github.com/theajk2/is310-coding-assignments/tree/main/pandas-eda

**Comment by ahmadsawwan** [comment]
https://github.com/ahmadsawwan/ahmadsawwan-is310-coding-assignments/tree/main/pandas-eda

**Reply by ZoeLeBlanc** [reply]
Hi @ahmadsawwan ! This is a great start but wanted to give some feedback. Some of these are just suggestions but others are fixes to some of the errors I was seeing.

First, in terms of how you're loading the data. While it's great you're using `requests` and `StringIO`, it's actually unnecessary with `pandas`. If you update your notebook with this code, you'll see it works without either of those libraries:

```python
url = "https://raw.githubusercontent.com/melaniewalsh/responsible-datasets-in-context/main/datasets/top-500-novels/library_top_500.csv"


# response = requests.get(url, verify=False)
# csv_data = StringIO(response.text)
df = pd.read_csv(url)
df
```
That's because Pandas automatically can parse csv or other data files from across the web!

Second issue is with your Altair graph that isn't rendering. I'm first going to show you a solution to your current code and then what would work in fewer lines of code.

Current code solution:

```python
top50_counts = top50['author_gender'].value_counts().reset_index().rename(columns={'author_gender': 'counts', 'index': 'author_gender'})
bot50_counts = bot50['author_gender'].value_counts().reset_index().rename(columns={'author_gender': 'counts', 'index': 'author_gender'})
bar_top50 = alt.Chart(top50_counts).mark_bar().encode(
    x=alt.X('author_gender:N', title='Gender'),
    y=alt.Y('counts:Q', title='Count'),
    color='author_gender:N'
).properties(
    title='Gender Distribution in Top 50 Rankings'
)
bar_top50
```
If you paste this code in your notebook, you'll see that the graph is finally working. What was happening was that you were trying to pass in the product of `.value_counts` without resetting the index or renaming the columns. If you just reset the index, you get `[index, author_gender]` as columns, but confusingly the index is the gender and author_gender is the counts. Which is why I'm chaining the `rename` function here to make the names match what's in the columns. But you'll notice the syntax is pretty hacky (I'm renaming author_gender and then naming another column author_gender, which is pretty confusing imo).

Instead, you could actually do this all directly with Altair with the following code:

```python
alt.Chart(top50).mark_bar().encode(
    x=alt.X('author_gender:N', title='Gender'),
    y=alt.Y('count(top_500_rank):Q', title='Count'),
    color='author_gender:N'
).properties(
    title='Gender Distribution in Top 50 Rankings'
)
```
Now you should see the exact same graph, but rather than using `value_counts`, we are using Altair's built-in `count` method on the `top_500_rank` column. Now neither solution is better but this second one is fewer lines of code and a bit more readable.

If you could fix your notebook with this code and take a crack at making one more graph then I'm happy to give you full marks. Hope my notes above help and if you're having issues please don't hesitate to reach out! Overall really good effort 👍

**Reply by ahmadsawwan** [reply]
Actually, I seem to have submitted the earlier version of my work, I fixed it but I guess I didn't push the right thing? I will now push the right one. please let me know if there is anything I need to change.

**Comment by kohtas2** [comment]
https://github.com/kohtas2/is310-coding-assignments/tree/main/pandas-eda

**Reply by jrfrye2** [reply]
Hi @kohtas2 

Your code looks good but you did not include a README or any discussion about the data. Your visualizations cover a wide range of topics, so it would be helpful to include information about what you hoped to find and what you are trying to draw attention to.

**Reply by kohtas2** [reply]
@jrfrye2 Thank you for the feedback. I added the readme file to the repo!

**Comment by adg77** [comment]
https://github.com/adg77/is310-coding-assignments/tree/main/pandas-eda

**Reply by ZoeLeBlanc** [reply]
Great work @adg77 ! Thought your focus on fantasy genre and author gender was great 👍

**Comment by jasonmorris8** [comment]
https://github.com/jasonmorris8/is310-coding-assignments/tree/main/pandas-eda

**Comment by AlanK3** [comment]
https://github.com/AlanK3/is310-coding-assignments/tree/main/pandas-eda

**Reply by jrfrye2** [reply]
Hi @AlanK3 

You did not include a README, but otherwise, everything looks great!

**Comment by elee33** [comment]
https://github.com/elee33/is310-coding-assignments/tree/main/pandas-eda

**Comment by officalcharles** [comment]
https://github.com/officalcharles/is310-coding-assignments/tree/master/pandas-eda

**Reply by jrfrye2** [reply]
Hi @officalcharles 

No errors with your code, but you did not use Altair for your visualization. It would have also been helpful if you had spent a little more time explaining the relevance and findings of the scatterplot or included more visualizations so that the point you are trying to make is very clearly illustrated.

**Comment by chuzesta** [comment]
https://github.com/chuzesta/is310-coding-assignments/blob/main/pandas-eda/GoodreadsRatingExplorationAndHistoricalTimeline.ipynb

**Comment by yukthas2** [comment]
https://github.com/yukthas2/is310-coding-assignments/tree/main/pandas_eda
